File: Assignment42/program3.c

//////////////////////////////////////////////////////////////////////
// File name :       program2.c
// Discription :     It is Recusive program which accept string from user 
//                   and count number of characters.
// Author :          Mangesh Ashok Bedre
// Date :            17/11/2025
//
//////////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////////
// Function Name :   Strlen
// Discription :     It is used to return count number of characters.
// Input  :          char *
// Output :          int
// Author :          Mangesh Ashok Bedre
// Date :            17/11/2025
//
//////////////////////////////////////////////////////////////////////

int Strlen(char * str)
{
    static int iCount = 0;
    
    if(*str == '\0')
    {
        return iCount;
    }
    else
    {
        iCount++;
        str++;
        Strlen(str);
    }   
        
}

int main()
{
    char ch[20] = {'\0'};
    int iRet = 0;

    printf("Enter the String : \n");
    scanf("%[^\n]s",ch);
    
    iRet = Strlen(ch);
    printf("Length of String is %d\n",iRet);

    return 0;
}

File: Assignment54/program3.java
// Write java program which accept array of characters from user and
// count number of capital characters.
// Input : b N j B R b A d G G
// Output : 6

import java.util.Scanner;

class MyArray
{
    public int CapArray(char Arr[])
    {
        int i = 0,iCount = 0;

        for(i = 0; i < Arr.length; i++)
        {
            if((Arr[i] >= 'A') &&(Arr[i] <= 'Z'))
            {
                iCount++;
            }
        }
        return iCount;
    }
}

class program3
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Size of array :");
        int iSize = sobj.nextInt();

        char Arr[] = new char [iSize];

        System.out.println("Enter elements in Arr : ");
        for(int i = 0; i < iSize; i++)
        {
            Arr[i] = sobj.next().charAt(0);
        }

        MyArray obj = new MyArray();
        int iRet = obj.CapArray(Arr);

        System.out.println("Number of capital alphabet : "+iRet);        
    }
}